lib/at_cmd_parser/src/at_utils.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+
+#include "at_utils.h"
+
+u32_t at_params_remove_spaces_from_beginning(char **str)
+{
+	if ((!str) || (!(*str))) {
+		return 0;
+	}
+
+	u32_t space_count = 0;
+
+	while (isspace(**str) && (**str)) {
+		space_count++;
+		(*str)++;
+	}
+	return space_count;
+}
+
+size_t at_params_get_cmd_length(const char *const str)
```

Comment:
You declare the `str` parameter as a constant pointer, but then cast it and copy it to a non-constant pointer to change it. I suggest to accept a non-constant pointer to begin with, and manipulate it directly.